lib/email_api/emailer.py

```diff
@@ -0,0 +1,147 @@
+import os
+from typing import Tuple, Dict, List, Optional
+from smtplib import SMTP_SSL
+from email.header import Header
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.utils import formatdate
+
+from structs.email_params import EmailParams
+from structs.smtp_account import SMTPAccount
+
+
+class Emailer:
+    """
+    Emailer class is responsible for building and sending emails.
+    """
+
+    # Holds absolute dirpath to directory where email attachements files are stored
+    EMAIL_ATTACHMENTS_ROOT_DIR = os.path.normpath(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "../email_attachments"
+        )
+    )
+
+    def __init__(self, smtp_account: SMTPAccount):
+        self._smtp_account = smtp_account
+
+    def send_multiple_emails(
+        self, emails: Dict[Tuple[str], EmailParams], as_html: bool = True
+    ):
+        """
+        send multiple emails
+        :param emails: (Dict) keys are list of email addresses, values are configured EmailParams dataclass
+        :param as_html: whether to send emails using html message body (if True), or plaintext (if False) (Default True)
+        """
+        for email_to, email_params in emails.items():
+            self.send_email(email_to, email_params, as_html)
+
+    def send_email(
+        self, email_to: Tuple[str], email_params: EmailParams, as_html: bool = True
```

Comment:
Building on the above, the data class could simply be

```
email_addresses: List[str]
```

And we just loop regardless if it's 1 or n